storage/innobase/handler/ha_innodb.cc

```diff
@@ -15904,13 +15904,42 @@ static int innobase_alter_encrypt_tablespace(handlerton *hton, THD *thd,
   /* If new tablespace definition says it's encrypted */
   bool is_new_encrypted = !Encryption::is_none(newenc.data());
 
-  if (!is_old_encrypted && is_new_encrypted) {
+  // It is only possible to mark tablespace to be skipped by encryption threads
+  // if it is not already encrypted.
+  if (!is_old_encrypted && !is_new_encrypted &&
+      alter_info->explicit_encryption) {
+    if (!fil_crypt_exclude_tablespace_from_rotation(space)) {
+      my_error(ER_EXCLUDE_ENCRYPTION_THREADS_RUNNING, MYF(0), space->name);
+      DBUG_RETURN(convert_error_code_to_mysql(DB_ERROR, 0, NULL));
+    }
+    DBUG_RETURN(error);
+  } else if (!is_old_encrypted && is_new_encrypted) {
+    // TODO: PS-5323 should disallow reaching this code if encryption threads
+    // are enabled. Thus it should be safe to remove the crypt_data here. Here
+    // we encrypt with MK, the tablespace was previously excluded from rotation.
+    if (space->crypt_data) {
```

Comment:
Here and rest of code, use space->crypt_data != nulltpr

Reply:
is 5323 done?

Reply:
Not yet. I was already implemented in PR: https://github.com/percona/percona-server/pull/3246, but although it was approved by Laurynas, it was never approved by you. default_table_encryption came along with tablespace encryption and now this PR needs to be rewritten.

Reply:
changed to space->crypt_data != nullptr